client/src/lib/attempt-result.js

```diff
@@ -256,18 +256,71 @@ export function autocompleteTimeAttemptResult(time) {
   return roundOver10Mins(time);
 }
 
+/**
+ * Checks if a given set of results is a new world record average and returns a corresponding boolean.
+ */
+export function isWorldRecordAverage(
+  attemptResults,
+  eventId,
+  officialWorldRecords
+) {
+  const newAverage = average(attemptResults, eventId);
+  const wrAverage = officialWorldRecords?.find(
+    (wr) => wr.type === 'average' && wr.event.id === eventId
+  )?.attemptResult;
+  // if no wrAverage is found for some reason (networking issue, brand new event, etc.),
+  // we get `number < undefined` which always evaluates to `false`
+  return newAverage <= wrAverage;
+}
+
+/**
+ * Checks if a given set of results contains a new world record single and returns a corresponding boolean.
+ */
+export function hasWorldRecordSingle(
+  attemptResults,
+  eventId,
+  officialWorldRecords
+) {
+  const completeAttempts = attemptResults.filter(isComplete);
+  const bestSingle = Math.min(...completeAttempts);
+  const wrSingle = officialWorldRecords?.find(
+    (wr) => wr.type === 'single' && wr.event.id === eventId
+  )?.attemptResult;
+  // if no wrSingle is found for some reason (networking issue, brand new event, etc.),
+  // we get `number < undefined` which always evaluates to `false`
+  return bestSingle <= wrSingle;
+}
+
 /**
  * Checks the given attempt results for discrepancies and returns
  * a warning message if some are found.
  */
-export function attemptResultsWarning(attemptResults, eventId) {
+export function attemptResultsWarning(
+  attemptResults,
+  eventId,
+  officialWorldRecords
+) {
   const skippedGapIndex =
     trimTrailingSkipped(attemptResults).indexOf(SKIPPED_VALUE);
   if (skippedGapIndex !== -1) {
     return `You've omitted attempt ${
       skippedGapIndex + 1
     }. Make sure it's intentional.`;
   }
+  const completeAttempts = attemptResults.filter(isComplete);
+  const bestSingle = Math.min(...completeAttempts);
+  const newWorldRecordSingle = hasWorldRecordSingle(
+    attemptResults,
+    eventId,
+    officialWorldRecords
+  );
+  if (newWorldRecordSingle) {
+    return `
+      The result you're trying to submit includes a new world record single
+      (${formatAttemptResult(bestSingle, eventId)}).
+      Please check that the results are accurate.
+    `;
+  }
```

Comment:
We want to wrap this in `if (completeAttempts.length > 0)`.

Reply:
I moved that line to a higher scope so now it includes the `333mbf` branch. Given my understanding of how the code works, I don't think this should affect behavior.